keps/sig-storage/625-csi-migration/README.md

```diff
@@ -156,10 +163,211 @@ Running this set of tests is optional for a per-provider basis. We would
+  - CSI migration will require CSI driver to be installed in the cluster so it can add CSI related API objects including CSIDriver, CSINode, VolumeAttachment. 
+  - The size of PV will include up to two more annotations including `volume.beta.kubernetes.io/storage-provisioner` and `volume.beta.kubernetes.io/migrated-to`. 
+  - The existing Node object will include new labels, specifically the CSI topology that are introduced by the CSI driver, e.g. `topology.gke.io/zone=us-central1-b` for GCE PD CSI Driver.
+  - For existing in-line volumes, there will be a new field under `VolumeAttachment.Spec.Source.VolumeAttachmentSource.InlineVolumeSpec` that will be populated if in-line volumes of migrated in-tree plugin is used.
+
+* **Will enabling / using this feature result in increasing time taken by any 
+operations covered by [existing SLIs/SLOs]?**
+  Depending on the design and implementation of the CSI Driver, the operation time taken could vary. 
+  In general, it might increase the total time spend because for the CSI sidecar to detect the object in the APIServer and do corresponding change through socket might add additional traffic compared to the in-tree plugin model.
+
+* **Will enabling / using this feature result in non-negligible increase of 
+resource usage (CPU, RAM, disk, IO, ...) in any components?**
+  It should not increasing the resource usage in a significant manner. But each CSI driver deployed on the node could take more CPU and RAM depending on the implementation.
+
+### Troubleshooting
+
+
+* **How does this feature react if the API server and/or etcd is unavailable?**
+
+CSI sidecars will not be able to monitor the status change of the API object. So all volume related operation will fail. The existing running container should not be impacted.
+When the feature is not enabled, only provision/deletion/resize should fail.
+
+* **What are other known failure modes?**
+  For each of them, fill in the following information by copying the below template:
+  - Bug in CSI driver or translation library.
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+      Operators should be able to identify issues regarding migrated PV when the two metrics `csi_sidecar_operation_seconds` and `storage_operation_duration_seconds` showed error ratio spike.
+    - Mitigation: What can be done to stop the bleeding, especially for already running user workloads?
+      Already running workload should not be impacted except when there is pod movement. To stop the bleeding, turn off the feature gate and bring the pod back to the node without CSI migration.
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Each CSI sidecars and drivers will have their own logging to help debug.
```

Comment:
On the kubelet side, you can look at kubelet logs too to see errors returned by the CSI drivers. PVC and Pod events also can show error messages.

Reply:
Done